dds/DCPS/transport/shmem/ShmemDataLink.cpp

```diff
@@ -233,7 +233,10 @@ ShmemDataLink::stop_i()
   {
     ACE_GUARD(ACE_Thread_Mutex, g, peer_alloc_mutex_);
     if (peer_alloc_) {
-      peer_alloc_->release(0 /*don't close*/);
+      if (peer_alloc_->release(1 /*close*/) == -1) {
```

Comment:
I haven't looked at the ACE PR in detail yet, but would having that fix remove the need to pass `1` here?

Reply:
I tested again, but a 1 is required, only than `release` on the underlying `ACE_Shared_Memory_Pool`, see https://github.com/DOCGroup/ACE_TAO/blob/5ea3ce0f676f3ae3d9347cf43b762d09c32745fd/ACE/ace/Malloc_T.inl#L65. That will release the shared memory from this process.

Reply:
OK, I need to review in more detail then.  There should be some distinction between detaching from a peer's allocator and completely removing one's own allocator.

Reply:
Only when called with `1` `ACE_Malloc_T` will call the underlying `release of `ACE_Shared_Memory_Pool`, the decision to destroy it is done based on the reference counting in the PI control block. With just `0` nothing happens to the `ACE_Shared_Memory_Pool`, looks the API between `ACE_Malloc_T` and `ACE_Shared_Memory_Pool` aren't 100% fit but we can't change that. 

Did just a test using `DevGuideExamples/DCPS/Messenger`, configured it to use shared memory, keep the subscriber running for some minutes and restarted the publisher multiple times, shared memory segment for the subscriber stays at 1, when the subscriber exists its own shared memory segment is also destroyed.

Reply:
Right, I didn't have the ACE code in front of me when I first reviewed it so it wasn't clear that the Malloc_T object treats this parameter differently.